resources/js/package.js

```diff
@@ -1,37 +1,55 @@
+import { set, useDebounceFn, useMemoize } from "@vueuse/core";
+
 document.addEventListener('turbo:load', function () {
+    window.app.$on('postcode-change', useDebounceFn(updateAddressFromPostcodeservice, 100));
+
     ['shipping_address', 'billing_address'].forEach((type) => {
         window.app.$watch('checkout.'+type+'.postcode', function () {
-            getAddressFromPostcodeservice(type)
+            window.app.$emit('postcode-change', window.app.checkout[type])
         })
 
         window.app.$watch('checkout.'+type+'.street.1', function () {
-            getAddressFromPostcodeservice(type)
+            window.app.$emit('postcode-change', window.app.checkout[type])
+        })
+
+        window.app.$watch('checkout.'+type+'.country_id', function () {
+            window.app.$emit('postcode-change', window.app.checkout[type])
```

Comment:
Do we really want to mix and match the different approaches like this? I feel like it'd be unintuitive to have some of these work without the `v-on:change` but other ones need it.